python/ray/_private/client_mode_hook.py

```diff
@@ -91,13 +107,23 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def client_mode_should_convert():
-    # This is for testing with RAY_CLIENT_MODE.
-    # When RAY_CLIENT_MODE=1, it means that for all the tests
-    # will run with client mode.
-    # is_client_mode_enabled will be set to be off when client is off
+def client_mode_should_convert(*, auto_init: bool):
+    """Determines if functions should be converted to client mode & if
+    Ray should be auto-initialized.
+
+    NOTE: `auto_init` must happen before we branch into regular ray or client
+    code because the initialization may result in either mode.
+    """
+    if auto_init:
+        import ray
+        if os.environ.get("RAY_ENABLE_AUTO_CONNECT",
+                          "") != "0" and not ray.is_initialized():
+            ray.init()
```

Comment:
cc @ericl @mwtian I just realized that changing from `ray.client().connect()` to `ray.init()` means that we will not check the `ray_current_cluster` file. 

I had to make some changes to workflows' tests because it depended on that behavior. Is this okay?